x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneMinMaxOperator.java

```diff
@@ -0,0 +1,175 @@
+
+    sealed interface NumberType permits LuceneMinFactory.NumberType, LuceneMaxFactory.NumberType {
+
+        /** Extract the competitive value from the {@link PointValues}  */
+        long fromPointValues(PointValues pointValues) throws IOException;
+
+        /** Wraps the provided {@link SortedNumericDocValues} with a {@link MultiValueMode} */
+        NumericDocValues multiValueMode(SortedNumericDocValues sortedNumericDocValues);
+
+        /** Return the competitive value between {@code value1} and {@code value2} */
+        long evaluate(long value1, long value2);
+
+        /** Build the corresponding block */
+        Block buildResult(BlockFactory blockFactory, long result, int pageSize);
+
+        /** Build the corresponding block */
+        Block buildEmptyResult(BlockFactory blockFactory, int pageSize);
+    }
+
+    private static final int PAGE_SIZE = 1;
+
+    private boolean seen = false;
+    private int remainingDocs;
+    private long result;
+
+    private final NumberType numberType;
+
+    private final String fieldName;
+
+    LuceneMinMaxOperator(
+        BlockFactory blockFactory,
+        LuceneSliceQueue sliceQueue,
+        String fieldName,
+        NumberType numberType,
+        int limit,
+        long initialResult
+    ) {
+        super(blockFactory, PAGE_SIZE, sliceQueue);
+        this.remainingDocs = limit;
+        this.numberType = numberType;
+        this.fieldName = fieldName;
+        this.result = initialResult;
+    }
+
+    @Override
+    public boolean isFinished() {
+        return doneCollecting || remainingDocs == 0;
+    }
+
+    @Override
+    public void finish() {
+        doneCollecting = true;
+    }
+
+    @Override
+    public Page getCheckedOutput() throws IOException {
+        if (isFinished()) {
+            assert remainingDocs <= 0 : remainingDocs;
+            return null;
+        }
+        final long start = System.nanoTime();
+        try {
+            final LuceneScorer scorer = getCurrentOrLoadNextScorer();
+            // no scorer means no more docs
+            if (scorer == null) {
+                remainingDocs = 0;
+            } else {
+                final LeafReader reader = scorer.leafReaderContext().reader();
+                final Query query = scorer.weight().getQuery();
+                if (query == null || query instanceof MatchAllDocsQuery) {
+                    final PointValues pointValues = reader.getPointValues(fieldName);
+                    // only apply shortcut if we are visiting all documents, otherwise we need to trigger the search
+                    // on doc values as that's the order they are visited without push down.
+                    if (pointValues != null && pointValues.getDocCount() >= remainingDocs) {
+                        final Bits liveDocs = reader.getLiveDocs();
+                        if (liveDocs == null) {
+                            // In data partitioning, we should have already collected it
```

Comment:
I think this is worth a longer comment.

I *believe* this happens because in data partitioning we might have got the same segment previous to this but with a different document range. And we're totally ignoring that range. We're just reading the min/max from the segment. That's sneaky, but it makes sense. And if we get another slice in the same segment we may as well skip it - we've already looked.

Some other thread *may* grab the a different slice of the same segment and, while that's not great, that's *fine* because it'll get the same min and max.